### PR TITLE
[dnscrypt-proxy] Creating chart for dnscrypt-proxy

### PR DESCRIPTION
**Pull request by obeone**
**Description of the change**
Add a new chart for dnscrypt-proxy. This stateless container proxy DNS request trough secured DNS protocol as dnscrypt, DoH, DoT.
It supports anonymized dnscrypt.

**Benefits**
More privacy. And easy to setup chart, allowing multiple replicas

**Possible drawbacks**
-

**Applicable issues**

- fixes #

**Additional information**

**Checklist** 
- [ X] Title of the PR starts with chart name (e.g. `[home-assistant]`)
- [ X] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [ X] Chart `artifacthub.io/changes` changelog annotation has been updated in `Chart.yaml`. See [Artifact Hub documentation](https://artifacthub.io/docs/topics/annotations/helm/#supported-annotations) for more info.
- [ X] Variables have been documented in the `values.yaml` file.

### CODE REVIEW

File: charts/stable/dnscrypt-proxy/templates/common.yaml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+{{- define "dnscryptproxy.configVolume" -}}
+enabled: "true"
+mountPath: "/config/dnscrypt-proxy.toml"
+subPath: "dnscrypt-proxy.toml"
+type: "custom"
+volumeSpec:
+  configMap:
+    name: {{ include "common.names.fullname" . }}
+{{- end -}}
+
```
Comment (bjw-s) [inline charts/stable/dnscrypt-proxy/templates/common.yaml:14]:
```suggestion
```

File: charts/stable/dnscrypt-proxy/templates/common.yaml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+{{- if .Values.config.enabled }}
+    {{/* To force pod restart on config update */}}
+    {{- $_ := set .Values.persistence "config" (include "dnscryptproxy.configVolume" . | fromYaml) -}}
+    {{- $_ := set .Values.podAnnotations "config/hash" (.Values.config.values | sha256sum ) -}}
+{{- end }}
+
```
Comment (bjw-s) [inline charts/stable/dnscrypt-proxy/templates/common.yaml:21]:
```suggestion
```

File: charts/stable/dnscrypt-proxy/values.yaml [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+# This chart inherits from our common library chart. You can check the default values/options here:
+# https://github.com/k8s-at-home/library-charts/tree/main/charts/stable/common/values.yaml
+#
+
+image:
+  # -- image repository
+  repository: klutchell/dnscrypt-proxy
+  # -- image tag
+  # @default -- chart.appVersion
+  tag: 2.1.1
```
Comment (bjw-s) [inline charts/stable/dnscrypt-proxy/values.yaml:13]:
```suggestion
  tag: 
```

**Review by bjw-s — CHANGES_REQUESTED** [review]
Hi! Thanks for taking the time to submit this PR!

I have left a few small textual comments, but the main change I want to propose is replacing the `config:` key with the `configmap:` key that is native to our common library. 

An example of this can be seen in the Frigate chart: 
- https://github.com/k8s-at-home/charts/blob/master/charts/stable/frigate/values.yaml#L99L120
- https://github.com/k8s-at-home/charts/blob/master/charts/stable/frigate/templates/common.yaml

File: charts/stable/dnscrypt-proxy/templates/common.yaml (hunk 1 above)
Comment (obeone) [reply charts/stable/dnscrypt-proxy/templates/common.yaml:14]: Oh... Yes, it's obvious !

I quickly correct it and push again

File: charts/stable/dnscrypt-proxy/values.yaml (hunk 3 above)
Comment (obeone) [reply charts/stable/dnscrypt-proxy/values.yaml:13]: Ok. I use everywhere I can k8s-at-home's charts, and I always saw fixed tag version, so I was thinking you prefer regular chart update but being sure nothing will break after a reboot for example and the download of a unexpected new version

Comment (obeone) [reply charts/stable/dnscrypt-proxy/values.yaml]: Sorry, I understood. It's already set to the Chart.yaml appVersion

**Pull request closed without merging**